Week 3

alsile opened the pull request.

alsile committed:
complete project 2

alsile committed:
update comments styles and formatting

huzcn commented on portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java:
```diff
@@ -0,0 +1,30 @@
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```
"datastore" can be created as a member variable so that you can need to recreate it in each post request. It can be initialized as:

private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();

huzcn commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -34,8 +35,8 @@
   
-  private List<String> output = new ArrayList<>();
-  
+  private Comments comments = new Comments(5, new ArrayList<String>());
```
"5" can be defined as a constant, e.g. private static final int COMMENT_LIMIT = 5.

huzcn commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -44,25 +45,35 @@
       }
     }
-
+    
```
remove unnecessary spaces

alsile replied:
let me know if I didnt do this, im not 100% sure but i think i did

huzcn commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -44,25 +45,35 @@
+    
+    String limit = request.getParameter("limit");
+    if (limit.equals("")) {
+      comments.limit = 5;
```
when you define a constant for limit, it can be directly used here.

huzcn reviewed (changes requested):
Added few comments in the first review.

alsile committed:
add translation feature

huzcn commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -48,33 +52,42 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
-    add.setProperty("timestamp", new Date());
+    if (!comment.equals("")) {
+      Entity add = new Entity("Task");
+      add.setProperty("title", comment);
```
It's better to give the property a meaningful name, e.g. "comment"

huzcn commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -48,33 +52,42 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
+                            Translate.TranslateOption.targetLanguage(comments.languageCode),
+                            Translate.TranslateOption.format("text"));
+      toDisplay += translation.getTranslatedText();
+      comments.jsonList.add(toDisplay);
```
Are you going to return all the comments in the datastore? Or only return comments.limit number of comments as the maximum?

huzcn commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -34,11 +37,12 @@
+  private Comments comments = new Comments(COMMENT_LIMIT_DEFAULT,
+                                           new ArrayList<String>(),
+                                           COMMENT_LANGUAGE_CODE_DEFAULT);
   private Gson gson = new Gson();
```
There is no need for this member variable. You can initialize it in where it's used below.

huzcn commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -48,33 +52,42 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
-    add.setProperty("title", comment);
-    add.setProperty("timestamp", new Date());
+    if (!comment.equals("")) {
+      Entity add = new Entity("Task");
```
I would give a meaningful name rather than "Task".

huzcn reviewed (changes requested):
Looks nice now! Added a few more comments.

alsile committed:
complete query method that passes all provided tests

alsile committed:
implement suggestions to improve variable names and to stop caching comments

alsile committed:
trying to get all changes on week 3

alsile committed:
complete a little over half of project 5, includding working query and added test, will add optional attendees functionality

alsile committed:
Revert "complete a little over half of project 5, includding working query and added test, will add optional attendees functionality"

This reverts commit c17c6eac88ee79396ee54e79b9c8b472ba2395d5.

alsile committed:
Revert "trying to get all changes on week 3"

This reverts commit e2e687757901fa05df3a659156ded6ac4be6aefc.

alsile committed:
Revert "complete query method that passes all provided tests"

This reverts commit cbc23ac93625ec86bf00cae1f23f1934be447582.

huzcn commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -33,37 +37,77 @@
+    }
+  }
+
+  private class CommentData {
```
It's better to move private members together, e.g. move this below "datastore".

huzcn commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -33,37 +37,77 @@
     response.sendRedirect("/index.html");
   }
+
+  private class Comment {
```
please move private members together

huzcn commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -33,37 +37,77 @@
-
-    response.setContentType("application/JSON;");
-    response.getWriter().println(gson.toJson(output));
+    
```
Remove empty spaces.

huzcn reviewed (approved):
Approve for merging after nits are resolved.

alsile committed:
Apply suggestions made in last PR

- remove empty spaces
- remove second set of Comment and CommentData class
- move first set of Comment and CommentData under other private elements
